atr/post/distribution.py

```diff
@@ -58,33 +57,14 @@ async def delete(session: web.Committer, project: str, version: str) -> web.Werk
 
 
 @post.committer("/distribution/record/<project>/<version>")
-async def record_post(session: web.Committer, project: str, version: str) -> str:
-    form = await shared.distribution.DistributeForm.create_form(data=await quart.request.form)
-    fpv = shared.distribution.FormProjectVersion(form=form, project=project, version=version)
-    if await form.validate():
-        return await shared.distribution.record_form_process_page(fpv)
-    match len(form.errors):
-        case 0:
-            # Should not happen
-            await quart.flash("Ambiguous submission errors", category="warning")
-        case 1:
-            await quart.flash("There was 1 submission error", category="error")
-        case _ as n:
-            await quart.flash(f"There were {n} submission errors", category="error")
-    return await shared.distribution.record_form_page(fpv)
+@post.form(shared.DistributeForm)
+async def record_post(session: web.Committer, form: shared.DistributeForm, project: str, version: str) -> str:
+    # Pydantic validation happens automatically in @post.form
+    return await shared.record_form_process_page_new(form, project, version, staging=False)
 
 
 @post.committer("/distribution/stage/<project>/<version>")
-async def stage_post(session: web.Committer, project: str, version: str) -> str:
-    form = await shared.distribution.DistributeForm.create_form(data=await quart.request.form)
-    fpv = shared.distribution.FormProjectVersion(form=form, project=project, version=version)
-    if await form.validate():
-        return await shared.distribution.record_form_process_page(fpv, staging=True)
-    match len(form.errors):
-        case 0:
-            await quart.flash("Ambiguous submission errors", category="warning")
-        case 1:
-            await quart.flash("There was 1 submission error", category="error")
-        case _ as n:
-            await quart.flash(f"There were {n} submission errors", category="error")
-    return await shared.distribution.record_form_page(fpv, staging=True)
+@post.form(shared.DistributeForm)
+async def stage_post(session: web.Committer, form: shared.DistributeForm, project: str, version: str) -> str:
```

Comment:
We don't call forms `form` because that conflicts with the `atr.form` module name when imported in our standard style. For `ExampleForm` we use the name `example_form`, so for `DistributeForm` that would be `distribute_form`.